src/preprocess.js

```diff
@@ -176,9 +176,10 @@ function preprocessPlantUmlIncludes (diagramText, dirPath, includeOnce, includeS
  */
 function resolveIncludeFile (includeFile, includePaths, vfs) {
   const exists = typeof vfs !== 'undefined' && typeof vfs.exists === 'function' ? vfs.exists : require('./node-fs.js').exists
+  const join = typeof vfs !== 'undefined' && typeof vfs.join === 'function' ? vfs.join : path.join
   let filePath = includeFile
   for (let i = 0; i < includePaths.length; i++) {
-    const localFilePath = path.join(includePaths[i], includeFile)
+    const localFilePath = join(includePaths[i], includeFile)
     if (exists(localFilePath)) {
```

Comment:
I think this part is wrong, we should resolve file using the file system (even when we are using Antora). The reason is that the PlantUML `!include` directive does not handle Antora resource IDs.

That's probably why you had to introduce `vfs.join`.

Path resolution is complicated especially when the context impacts how we should resolve a path.

Reply:
This is there for the case where we join the antora resource dirname (e.g. "partial$architecture") with the included relative path (e.g. "ba.puml"), where Antora requires forward slashes in resource ids (i.e. "partial$architecture/ba.puml" and not "partial$architecture\ba.puml").

Reply:
What you're saying is completely rewriting the way the Antora resources are loaded. Honestly, I'm not sure that's worth the effort, and it's not an effort I can afford to put into this right now. I can close the pull request and just continue running off of our own fork if you like.

Reply:
Are you using Antora resource IDs in PlantUML files with the `!include` directive?

Reply:
No, I'm using relative paths in !include directives. See the test case at https://github.com/Mogztter/asciidoctor-kroki/pull/396/files#diff-f387f1c7527cf78a5c58201fcd970bbaaa8dbd7a9dbfb9b51e58b69f589226ac

Reply:
Looking further at the Antora documentation, I don't see any other way to do it. https://docs.antora.org/antora/latest/organize-content-files/#antoras-virtual-filesystem states:
> The only aspect of a file that maps back to the location on the filesystem is the family-relative path

Reply:
I will take a deeper look later this week, maybe it's more complicated than what I have in mind.

Reply:
@ggrossetie Any update on this?